x-pack/plugins/ml/public/maps/update_anomaly_source_editor.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Fragment, Component } from 'react';
+
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiPanel, EuiSpacer, EuiTitle } from '@elastic/eui';
+import { LayerSelector } from './layer_selector';
+import { MlAnomalyLayers } from './util';
+
+interface Props {
+  onChange: (...args: Array<{ propName: string; value: unknown }>) => void;
+  typicalActual: MlAnomalyLayers;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface State {}
+
+export class UpdateAnomalySourceEditor extends Component<Props, State> {
+  state: State = {};
+
+  render() {
+    return (
+      <Fragment>
+        <EuiPanel>
+          <EuiTitle size="xs">
+            <h6>
+              <FormattedMessage id="xpack.ml.maps.settingsEditorLabel" defaultMessage="Anomalies" />
+            </h6>
+          </EuiTitle>
+          <EuiSpacer size="s" />
+          <LayerSelector
+            onChange={(typicalActual: MlAnomalyLayers) => {
+              this.props.onChange({
+                propName: 'typicalActual',
+                value: typicalActual,
+              });
+            }}
+            typicalActual={this.props.typicalActual}
+          />
+        </EuiPanel>
+        <EuiSpacer size="s" />
```

Comment:
Would it be useful to add a switch here to turn on/off "time-awareness".

E.g. something along the lines of:

![image](https://user-images.githubusercontent.com/1833023/150019787-f61a6575-2c8f-46ac-8517-ee7219c23675.png)

E.g. add another property `applyGlobalTime` to the source-descriptor.

```
export interface AnomalySourceDescriptor extends AbstractSourceDescriptor {
  jobId: string;
  typicalActual: MlAnomalyLayers;
  applyGlobalTime: boolean;
}
```

The switch controls the value of `applyGlobalTime`.

Then implement `AnomalySource#isTimeAware` as:

```
  async isTimeAware(): Promise<boolean> {
    return this._descriptor.applyGlobalTime;
  }
```

This way you can turn on/off whether users will see _all_ the data (regardless of time-selection), or only a cut of the data.

Reply:
Sounds like a good one to discuss for a follow-up 👍